src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -102,17 +105,25 @@ mod imp {
         let addr = siginfo_si_addr(info);
 
         // If the faulting address is within the guard page, then we print a
-        // message saying so.
+        // message saying so and abort.
         if guard != 0 && guard - PAGE_SIZE <= addr && addr < guard {
             report_overflow();
-        }
 
-        // Unregister ourselves by reverting back to the default behavior.
-        let mut action: sigaction = mem::zeroed();
-        action.sa_sigaction = SIG_DFL;
-        sigaction(signum, &action, ptr::null_mut());
+            // Unregister any existing abort handler to get the default behavior.
+            let mut action: sigaction = mem::zeroed();
+            action.sa_sigaction = SIG_DFL;
+            sigaction(libc::SIGABRT, &action, ptr::null_mut());
 
-        // See comment above for why this function returns.
+            libc::raise(libc::SIGABRT);
```

Comment:
Would it make more sense to change `rtabort!` to call `libc::abort` instead of `intrinsics::abort`, now that we have `libc::abort`? `libc::abort` is supposed to handle all of the edge cases like this, either unregistering the signal handler or killing itself another way if the signal handler returns or just infinitely looping if all else fails.